src/sparseml/exporters/transforms/kv_cache/transforms_base.py

```diff
@@ -97,6 +98,76 @@ def find_nodes_by_pattern(
             raise ValueError(f"Unable to find pattern:\n{pattern}\nin model")
         return [match.node for match in matches]
 
+    def adjust_causal_mask(
+        self,
+        model: ModelProto,
+        nodes_to_add: List[NodeProto],
+        initializers_to_add: Optional[List[TensorProto]] = None,
+    ) -> ModelProto:
+        """
+        Insert series of nodes after the causal mask input to change
+        the initial int64 mask, to a mask expected by the model.
+
+        Transform:
+        ```
+        |       causal_mask
+        |            |
+        |   causal_mask_input_child
+        ```
+        to:
+        ```
+        |       causal_mask
+        |            |
+        |       nodes_to_add[0]
+        |            |
+        |       nodes_to_add[1]
+        |            |
+        |           ...
+        |            |
+        |       nodes_to_add[k]
+        |            |
+        |            |
+        |   causal_mask_input_child
+        ```
+
+        :param model: the ONNX model
+        :param nodes_to_add: a list of nodes to add after the causal mask
+            input. The ordering of nodes is important. The first node in the list
+            must take the causal mask as input. The last node in the list will be
+            connected to the `causal_mask_input_child`.
+        :param initializers_to_add: an optional list of initializers to add to the model
+        :return: model with the causal mask adjusted
+        """
+
+        graph = ONNXGraph(model)
+
+        if nodes_to_add[0].input[0] != self.CAUSAL_MASK_NAME:
```

Comment:
lgtm overall - let's discuss tomorrow if there's a way we can clean up this flow a bit. needing to format this list arg in a certain way seems a bit like hidden knowledge here